domains/auth/Auth.API/API/Services/ICryptographyService.cs

```diff
@@ -2,6 +2,7 @@ namespace API.Services;
 public interface ICryptographyService
 {
     public string Encrypt(string state);
```

Comment:
Why is the signature not this for Encrypt?

```suggestion
    public string Encrypt<T>(T state);
```

That way we will be able to encrypt and decrypt AuthState.